subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -457,15 +454,20 @@ static void http_fota_handler(enum fota_download_evt_id evt)
 {
 	__ASSERT_NO_MSG(c != NULL);
 
+	int err = 0;
+
 	switch (evt) {
 	case FOTA_DOWNLOAD_EVT_FINISHED:
 		fota_state = APPLY_FIRMWARE;
-		update_job_execution(c, job_id, AWS_JOBS_IN_PROGRESS,
+		err = update_job_execution(c, job_id, AWS_JOBS_IN_PROGRESS,
 				     fota_state, doc_version_number, "");
+		if (err != 0) {
+			callback(AWS_FOTA_EVT_ERROR);
+		}
 		break;
 	case FOTA_DOWNLOAD_EVT_ERROR:
 		LOG_ERR("FOTA download failed, report back");
-		update_job_execution(c, job_id, AWS_JOBS_FAILED,
+		(void) update_job_execution(c, job_id, AWS_JOBS_FAILED,
```

Comment:
instead of casting it to void assign it and put it in the log err msg.

Reply:
It's already printed at line 217.

Reply:
oki